src/Qir/Runtime/public/QirRuntime.hpp

```diff
@@ -66,7 +53,7 @@ extern "C"
     QIR_SHARED_API char* quantum__rt__memory_allocate(uint64_t size); // NOLINT
 
     // Fail the computation with the given error message.
-    QIR_SHARED_API void quantum__rt__fail(QirString* msg); // NOLINT
+    [[noreturn]] QIR_SHARED_API void quantum__rt__fail(QirString* msg); // NOLINT
```

Comment:
I'm not familiar with the `[[noreturn]]` decoration. Does that indicate that the function unconditionally throws exception or asserts or otherwise terminates? Seems handy!

Reply:
See [here](https://en.cppreference.com/w/cpp/language/attributes/noreturn).
I believe it affects optimization and likely something else.